go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -542,6 +524,58 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 	return nil
 }
 
+// hasIPAMClaim determines whether a pod's IPAM is being handled by IPAMClaim CR.
+// pod passed should already be validated as having a network connection to nadName
+func (bsnc *BaseSecondaryNetworkController) hasIPAMClaim(pod *kapi.Pod, nadNamespacedName string) (bool, error) {
+	if !bsnc.AllowsPersistentIPs() {
+		return false, nil
+	}
+
+	var ipamClaimName string
+	var wasPersistentIPRequested bool
+	if bsnc.IsPrimaryNetwork() {
+		// primary network ipam reference claim is on the annotation
+		ipamClaimName, wasPersistentIPRequested = pod.Annotations[util.OvnUDNIPAMClaimName]
+	} else {
+		// secondary network the IPAM claim reference is on the network selection element
+		nadKeys := strings.Split(nadNamespacedName, "/")
+		if len(nadKeys) != 2 {
+			return false, fmt.Errorf("invalid NAD name %s", nadNamespacedName)
+		}
+		nadNamespace := nadKeys[0]
+		nadName := nadKeys[1]
+		allNetworks, err := util.GetK8sPodAllNetworkSelections(pod)
```

Comment:
@maiqueb I think we should do a FUP for adding tests for IPAM claims deletion, the fact that we were not handling all the networks and we didn't catch it looks like its not being tested? wanna open an issue?

Reply:
What do you mean we weren't handling all the networks ? 

And we have tests deleting the allocations (... on the cluster manager side ...): 
- https://github.com/ovn-kubernetes/ovn-kubernetes/blob/d1ebf1c6e335cf095458a3beafd1a601d3eb5a71/go-controller/pkg/clustermanager/pod/allocator_test.go#L452
- https://github.com/ovn-kubernetes/ovn-kubernetes/blob/d1ebf1c6e335cf095458a3beafd1a601d3eb5a71/go-controller/pkg/clustermanager/pod/allocator_test.go#L475

This actually leads me to question: isn't this part of the code only used on non-IC ?...

Reply:
I think by "all" she means primary and secondary networks with network selection elements.